src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/PropertyPages/LaunchProfileFromProjectDataProducer.cs

```diff
@@ -0,0 +1,119 @@
+                && await launchSettingsProvider.WaitForFirstSnapshot(Timeout.Infinite) is ILaunchSettings launchSettings)
+            {
+                return createLaunchProfileValues();
+            }
+
+            return Enumerable.Empty<IEntityValue>();
+
+            IEnumerable<IEntityValue> createLaunchProfileValues()
+            {
+                IPropertyPageQueryCache propertyPageQueryCache = _queryCacheProvider.CreateCache(project);
+
+                Dictionary<string, Rule> debugRules = new();
+                foreach (Rule rule in DebugUtilities.GetDebugChildRules(projectCatalog))
+                {
+                    if (rule.Metadata.TryGetValue("CommandName", out object? commandNameObj)
+                        && commandNameObj is string commandName)
+                    {
+                        debugRules[commandName] = rule;
+                    }
+                }
+
+                foreach ((int index, ProjectSystem.Debug.ILaunchProfile profile) in launchSettings.Profiles.WithIndices())
+                {
+                    if (!Strings.IsNullOrEmpty(profile.Name)
+                        && !Strings.IsNullOrEmpty(profile.CommandName)
+                        && debugRules.TryGetValue(profile.CommandName, out Rule rule))
+                    {
+                        QueryProjectPropertiesContext context = new(
+                            isProjectFile: true,
+                            file: project.FullPath,
+                            itemType: LaunchProfileProjectItemProvider.ItemType,
+                            itemName: profile.Name);
+
+                        IEntityValue launchProfileValue = CreateLaunchProfileValue(parent, context, rule, index, propertyPageQueryCache);
+                        yield return launchProfileValue;
+                    }
+                }
+            }
+        }
+
+        private IEntityValue CreateLaunchProfileValue(IEntityValue parent, QueryProjectPropertiesContext context, Rule rule, int order, IPropertyPageQueryCache propertyPageQueryCache)
+        {
+            EntityIdentity identity = new(
+                ((IEntityWithId)parent).Id,
+                new Dictionary<string, string>
+                {
+                    { ProjectModelIdentityKeys.SourceItemType, context.ItemType! },
+                    { ProjectModelIdentityKeys.SourceItemName, context.ItemName! }
+                });
+
+            return CreateLaunchProfileValue(parent.EntityRuntime, identity, context, rule, order, propertyPageQueryCache);
+        }
+
+        private IEntityValue CreateLaunchProfileValue(IEntityRuntimeModel runtimeModel, EntityIdentity id, QueryProjectPropertiesContext context, Rule rule, int order, IPropertyPageQueryCache cache)
+        {
+            LaunchProfileValue newLaunchProfile = new(runtimeModel, id, new LaunchProfilePropertiesAvailableStatus());
+
+            if (_properties.Name)
```

Comment:
What are these predicates? Is it some kind of new implicit null check I'm not aware of? I see them here and more in other places where the if-expression is simply a non-boolean member.

Reply:
The `ILaunchProfilePropertiesAvailableStatus` type (and more generally all `*AvailableStatus` types) are used by the Projcet Query API to indicate which parts of the model should be populated. `_properties.Name` is in fact a boolean; `LaunchProfileValue` and `ILaunchProfilePropertiesAvailableStatus` have the same member names, but all of those members are booleans on the latter.